### PR TITLE
[CARBONDATA-36]fix dictionary exception when data column count less then schema

### PR DESCRIPTION
**Pull request by eason-lyx**
when csv head is id,name,age,city
if the data record like as the following:
1,a,10,china
2,b
3,c,11,usa
then,when read this row:2,b,generating dicionary will failure,and the remain record (3,c,11,usa)will no generate dictionary.

this pr to prevent put null value to Set,and let it continue.

### CODE REVIEW

File: integration/spark/src/main/scala/org/carbondata/spark/rdd/CarbonGlobalDictionaryRDD.scala [hunk 1]
```diff
@@ -67,7 +67,9 @@ case class PrimitiveParser(dimension: CarbonDimension,
 
   def parseString(input: String): Unit = {
     if (hasDictEncoding) {
-      set.add(input)
+      if (StringUtils.isNotEmpty(input)) {
```
Comment (kumarvishal09) [inline integration/spark/src/main/scala/org/carbondata/spark/rdd/CarbonGlobalDictionaryRDD.scala]: empty value in data is a valid scenario , here as we are using hashtable we cannot insert null value i think null check is ok..Please correct me if i am worng

Comment (eason-lyx) [inline integration/spark/src/main/scala/org/carbondata/spark/rdd/CarbonGlobalDictionaryRDD.scala]: yes empty value is a valid scenario.
in dictionary file,we already contain the null value.so here we no need to insert to set,also set no allow insert null.so dictionary generate fail in this place.
and in csv reader:we get the row like:1,abc,null,null,null
we can match the null value in dictionary file.

Comment (kumarvishal09) [inline integration/spark/src/main/scala/org/carbondata/spark/rdd/CarbonGlobalDictionaryRDD.scala]: please update the code

**Comment by Vimal-Das** [comment]
LGTM

**Comment by Vimal-Das** [comment]
Merged to master.

**Pull request merged**